articles/sql-user-defined-functions/index.md

```diff
@@ -0,0 +1,266 @@
+
+# Working With User-Defined Functions In SQL Server
+
+  
+
+In this tutorial, we will be discussing user-defined functions in SQL Server. More specifically, we will be discussing Scalar Functions and Table-Valued Functions.
+
+When writing code, one must aim to follow the DRY Principle (Don’t Repeat Yourself). One way to avoid repetition of code is to put chunks of code inside Functions and invoke them as required.
+
+
+The concept of Functions in SQL is like that in any other programming language like Python. The major difference being the way they are implemented. There are two main types of user-defined functions in SQL based on the data they return
+
+  
+
+- **Scalar Functions-** These types of Functions return a single value, i.e float, int, varchar, datetime, etc
+
+- **Table-Valued Functions-** These Functions return tables
+
+  
+
+### Table of contents
+
+- Prerequisites
+
+- Creating Functions
+
+- Using Functions in Statements
+
+- Update/Delete Functions
+
+- Using Variables, Conditional Statements inside Functions
+
+- Conclusion
+
+  
+
+### Prerequisites
+
+  
+
+- A basic understanding of SQL
+
+  
+
+- A SQL Server with a database
+
+- [SQL Server Management Studio](https://docs.microsoft.com/en-us/sql/ssms/download-sql-server-management-studio-ssms?view=sql-server-ver15) to connect to your database.
+
+---
+
+### Creating Functions
+
+#### Scalar Functions
+
+Below is the definition of a simple Function. It takes in two numbers and returns their sum. Since this function returns a number, it is a scalar function.
+```sql
+    CREATE FUNCTION scalar_func
```

Comment:
It would be beneficial to beginners if you could add code comments for all the code that you've used in this article.